milli/src/update/settings.rs

```diff
@@ -283,6 +298,55 @@ impl<'a, 't, 'u, 'i> Settings<'a, 't, 'u, 'i> {
         }
     }
 
+    fn update_synonyms(&mut self) -> anyhow::Result<bool> {
+        match self.synonyms {
+            Setting::Set(ref synonyms) => {
```

Comment:
In this block could you please move the `normalize` closure first, then the `analyzer` configuration followed by the `new_synonyms` creation (with normalization), please? :)